docs/source/integrations/plugins.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Plugins
+description: Extend Apollo Server with custom functionality
+---
+
+> Plugins are available in Apollo Server 2.2.x and later.
+
+**Plugins** enable you to extend Apollo Server's core functionality by performing 
+custom operations in response to certain events. Currently, these events correspond
+to individual phases of the GraphQL request lifecycle, and to the startup of Apollo Server itself.
+
+For example, a basic logging plugin might log the GraphQL query string associated 
+with each request that's sent to Apollo Server.
+
+## Creating a plugin
+
+Plugins are JavaScript objects that implement one or more functions that respond to 
+events. Here's a basic plugin that responds to the `serverWillStart` event:
+
+```js:title=index.js
+const myPlugin = {
+  serverWillStart() {
+    console.log('Server starting up!');
+  },
+};
+```
+
+> If you're using TypeScript to create a plugin, the `apollo-server-plugin-base` module exports the [`ApolloServerPlugin` interface](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-plugin-base/src/index.ts) for plugins to implement.
+
+You can define a plugin in the same file where you initialize Apollo Server, or
+you can export it as a separate module:
+
+```js:title=myplugin.js
+module.exports = {
+  serverWillStart() {
+    console.log('Server starting up!');
+  },
+};
+```
+
+To create a plugin that accepts options, create a function that accepts an
+`options` object and returns a properly structured plugin object, like so:
+
+```js:title=myplugin.js
+module.exports = (options) => {
+  return {
+    serverWillStart() {
+      console.log(options.logMessage);
+    },
+  };
+};
+```
+
+
+### Responding to events
+
+A plugin specifies exactly which [lifecycle events](#plugin-event-reference)
+it responds to by implementing functions that correspond to those events.
+The plugin in the examples above responds to the `serverWillStart` event, which
+fires when Apollo Server is preparing to start up.
+
+A plugin can respond to any combination of supported events.
+
+#### Responding to request lifecycle events
+
+Plugins can respond to the following events associated with the GraphQL request
+lifecycle:
+
+* [`parsingDidStart`](#parsingdidstart)
+* [`validationDidStart`](#validationdidstart)
+* [`didResolveOperation`](#didresolveoperation)
```

Comment:
`responseForOperation` should also be included, shouldn't it? I think it was added later only a while ago, [see here](https://github.com/apollographql/apollo-server/blame/master/packages/apollo-server-plugin-base/src/index.ts#L50-L60).